### DISCUSSION TITLE
Any good user queries/filters for weeding out clickbait?

### DISCUSSION THREAD

**Discussion by yllekz** [FreshRSS/FreshRSS#6649] [General]
Hi folks, I've been experimenting with user queries to try to sniff out articles that are AI-generated or low quality clickbait. I've been trying to cobble one together on my own but was curious if anyone had anything better than what I came up with.

Here's my "low quality" filter so far:
```
intitle:'weird trick' or intitle:'simple trick' or intitle:'deals' or intitle:'sales' or intitle:'what we know' or intitle:'what to know' or intitle:"here's" or intitle:'everything new' or intitle:'these' or intitle:'?' or intitle:'and more' or intitle:'you need to know' or intitle:'this' or intitle:'but then'
```

**Comment by Frenzie** [comment]
My weird trick is not to subscribe to websites with such headlines. :innocent: 

But just the word "this"?

**Reply by yllekz** [reply]
You'd be surprised how well that one works for me actually. Yeah it won't be 100% but I've found that a lot of titles with "this" end up trying to be a misleading or clickbaitey article.

**Reply by Frenzie** [reply]
Yes, I see what you mean. The weird trick is probably "this one weird trick," while more legitimately it's rarer. I still see plenty though, for example https://xkcd.com/2879/

**Comment by Alkarex** [comment]
I have an *Uninteresting* user query, which I use to quickly look at typically uninteresting articles (for me) before marking them as read:

```#Bons+Plans OR intitle:Amazon OR intitle:Apple OR intitle:iPhone OR intitle:Samsung OR intitle:Xiaomi OR intitle:OnePlus OR intitle:Oracle OR intitle:€ OR intitle:$ OR intitle:🔥 OR intitle:"Bon plan" OR intitle:Instagram OR intitle:TikTok OR intitle:Netflix OR intitle:HBO OR intitle:Disney OR intitle:rugby OR intitle:football OR intitle:fodbold OR intitle:handball OR intitle:cyclisme OR intitle:gaming OR intitle:game OR intitle:Nintendo OR intitle:Sony OR intitle:PS5 OR intitle:Xbox OR intitle:promotion OR intitle:journée OR intitle:champion```

By the way, I have on my TODO to make the `intitle:` filter accept multiple alternatives, in order to be offer a shorter syntax

**Comment by Alkarex** [comment]
Related: AI classification implemented in:
* https://github.com/FreshRSS/Extensions/pull/458

**Comment by zapovit** [comment]
Check this one out, works quite well for Betteridge's law of headlines while preserving who/what/when/where/why/how "content" questions:
```
intitle:/^(?!.*(wh|how)).*\?\s*$/i
```
Applied to any kind of "news" category I find it captures all Betteridge-type headlines with a very low rate of false positive. 

AI (freetext)
```
/A\.?I\.?\b/ OR /Palantir/ OR /Anthropic/ OR /OpenClaw/i OR /artificial intelligence/i
```

**Reply by zapovit** [reply]
Improved AI (freetext):
```
/A\.?I\.?\b/ OR /Palantir/ OR /Anthropic/ OR /DeepMind/ OR (/Hugging Face/) OR /Cohere/ OR /TPU/ OR /nvidia/i OR /Groq/ OR /Tenstorrent/i OR /HBM/ OR /OpenClaw/i
```
This is more for companies than specific models

**Comment by mtalexan** [comment]
I have a few different User Queries I defined for different contexts of "spam" or things that don't interest me, and then I use those queries in the Mark as Read filter for a number of different individual feeds. I have a specific Spam filter, but I usually use it in context with some of the others as well (like Cooking from news feeds).

The biggest thing is looking for non-third-person phrases on top of the standard ones in my Spam list. If they're saying "us" or "our" or "you", that's not how most factual reporting is written so it is likely spam. But if you're looking at blog posts they often do use these phrases so you wouldn't want to exclude them.

⚙️Cooking:
```
(inurl:/(?<!tim )\bcook(ing)?/i) OR #/cook(ing)?/i OR #recipe OR #grill OR #condiments OR #sauce OR (#"Food & Drink") OR (#"Food News")
```

⚙️Art:
```
(intitle:/(?<!\b(to) )(?<!\b(for|the) )(?<!\b(help) )\bart(ist(s|ic))\b/i) OR #/\bart(ist(s|ic)?|s)?\b/i OR (intitle:"AP Photos") OR (intitle:"photo collection") OR (intitle:"photos of the week") OR (intitle:"photo book") OR intitle:"photos:" OR #/(?<!vis-)photo(s|graph)/i OR (intitle:/\bin pictures\b/i) OR (intitle:/^In Images\b/i) OR inurl:/\/arts?\//i
```

⚙️Death:
```
(intitle:/\bdies($| at\b)/i) OR (intitle:/\bdead at\b/i) OR #/obituar(y|ies)/i OR #/^deaths$/i OR (intitle:/\btribute to\b/i) OR intitle:/remember(ed|ing)/i OR intitle:/\bRequiem\b/i OR #/\brequiem\b/i
```

⚙️Financial:
```
(intitle:/\b(board|chairman) of\b/i) OR intitle:/\bequity\b/i OR intitle:/\b(financ(e|ial)s?|quarters?|stocks?|bonds?|markets|futures)\b/i OR (intitle:/\b(stock|bond) market\b/i) OR #finance OR #financial OR #stock OR #bond OR #market OR #futures OR (intitle:/(first|second|third|fourth) quarters?/i) OR (intitle:/\bfunding round\b/i) OR (intitle:/\b(net|gross) sales/i) OR (intitle:/\b(revenue|stake|valuation|valued at)\b/i) OR (intitle:/\bseries [A-Z]\b/i) OR (intitle:/\b(up|down|rise(s| of)|falls?) [0-9]+%/i) OR (intitle:/\byear (over|on) year\b/i) OR intitle:/\bYoY\b/i OR intitle:/\by\/y\b/i OR intitle:/\bIPO\b/ OR (intitle:/[$€£][0-9.]+( per |\/)share\b/i) OR (intitle:/\bstock buyback\b/i) OR (intitle:/([$€£]|&pound;)[0-9.]+(M|B)? ((funding|series) )?round\b/i) OR intitle:/\bunion/i OR #/\bunion/i OR intitle:/\bcapex\b/i OR (intitle:/\bcapital expenditure/i) OR (intitle:/(?!inflation )\bexpectations\b/i) OR intitle:/\bprofits?\b/i
```

⚙️InReview:
```
(intitle:/\bago today\b/i) OR (intitle:/(decades?|years?|months?|quarters?) (ago|past|on)\b/i) OR (intitle:/([0-9]+|reviewing|remembering) (decades?|years?|months?|quarters?)/i -intitle:/\b(given|sentenced|condemned)\b/i) OR (intitle:/\b(on|from|in) 20[0-2][0-9]/i) OR (intitle:/\bin review\b/i) OR intitle:/\bremember(s|ed|ing)\b/i OR (intitle:/\breview (of )?the (month|year|week)/i) OR (intitle:/(^|\bon )this (day|week|month)\b/i)
```

⚙️Interview:
```
intitle:/^Meet\b/i OR (intitle:/\bopens up about\b/i) OR (intitle:/\b([A-Z][a-z]+ ){2}(?i)(explains|discusses|jokes|chats)\b/ -intitle:/\b(judge|courts?|justice|client|president|prime minister)\b/i) OR (intitle:/\bchat about\b/i)
```

⚙️MeetsVisit:
```
intitle:/\bme(t|ets)\b/i OR (intitle:/\b(to|will) meet\b/i) OR intitle:/\bvisit(s|ed)\b/i OR (intitle:/\b(to|will) visit\b/i) OR (intitle:/\barrives (in|at)\b/i) OR (intitle:/\bwill arrive (in|at)\b/i) OR (intitle:/\b will head to\b/i) OR (intitle:/\bhead(s|ed) to\b/i)
```

⚙️Opinion:
```
inurl:/\/opinions?\//i OR #opinion OR intitle:/\bopinion\b/i
```

⚙️Podcast:
```
inurl:/\/podcast(s|er)?\//i OR #podcast OR intitle:/\bpodcast(s|er)\b/i
```

⚙️Poll:
```
intitle:/^poll\b/i OR intitle:/\bQ&A\b/i OR #poll OR #survey OR (intitle:"readers have spoken") OR (intitle:/\btell us\b/i)
```

⚙️Spam:
```
(intitle:/[0-9]+ (things|tips)\b/i) OR intitle:/\b(deals|sale|save|discount)\b/i OR (intitle:/\bfor a limited time/i) OR (intitle:/\b(to|only|at|for) ([$][0-9]+|[0-9]+%)/i -intitle:/\btotal to\b/i) OR (intitle:/([$][0-9]+|[0-9]+%) (off|discount)\b/i) OR intitle:/\bvs[.]?\b/i OR intitle:/\bbest\b.*\bvpn\b/i OR (intitle:/\bbuy right now\b/i) OR #discount OR #sale OR #sponsor OR #endorse OR #save OR #deal OR (intitle:/\b(discounted|down|save|reduced) [0-9]%/i) OR (intitle:/\bright now\b/i) OR (intitle:/^(the )?best\b/i) OR intitle:/\btips\b/i OR (intitle:/\btop picks\b/i) OR (intitle:/^best\b(?! buy\b)/i) OR (intitle:/\bneed to know\b/i) OR (intitle:/\b(low(er|est|ed)|discounted|reduced|for (only)?|at|as little as|down to) [$][0-9]+/i)
```

⚙️Sports:
```
(intitle:/\b(N[.]?(F|H)[.]?L[.]?|N[.]?B[.]?A[.]?|M[.]?L[.]?B[.]?|N[.]?C[.]?A[.]?A[.]?)\b(?! legend )/i -intitle:/\b(judge|case|indicted|indictment)\b/i) OR (intitle:/(Europa|Premier) League/i) OR inurl:"/sports/" OR #/\bsports?\b/i OR (#/(Europa|Premier) League/i) OR inurl:"/soccer/" OR inurl:"/football/" OR inurl:"/baseball/" OR inurl:"/tennis/" OR inurl:"/golf/" OR #soccer OR #football OR #baseball OR #tennis OR #golf
```

⚙️IorMy:
```
intitle:/\b(my|I)\b(?!(.*?((\W['"])|(["'](\W|$)))))/i
```

⚙️Timeline:
```
intitle:/\btimeline\b/i
```

⚙️History:
```
(intitle:/today in history/i) OR (intitle:/\bon this day\b/i) OR (intitle:/\byears? on\b/i) OR (intitle:/\byear in review\b/i) OR (intitle:/\breview of the year\b/i)
```

⚙️You:
```
intitle:/\byour?\b(?!(.*?((\W['"])|(["'](\W|$)))))/i
```
 
⚙️Entertainment:
```
#/\bmovies\b/i OR #/\bentertainment\b/i OR inurl:"/style/" OR inurl:"/books/" OR inurl:"/theater/" OR inurl:"/television/" OR inurl:"/movies/"
```
    
⚙️UsorOur:
```
intitle:/\b(our|us)\b(?!(.*?((\W['"])|(["'](\W|$)))))/i
```

⚙️AI:
```
(#/(Artificial|Machine) Intelligence/i) OR (intitle:/(Artificial|Machine) Intelligence/i) OR intitle:/\bGrok\b/ OR intitle:/\bGemini\b/ OR intitle:/\bCopilot\b/ OR (intitle:/\bMachine Learning\b/i) OR intitle:/\bLLM\b/ OR intitle:/\bChatGPT\b/i OR (intitle:/\blarge language model\b/i) OR (intitle:/\bdeep learning\b/i) OR (intitle:/\bneural net\b/i) OR intitle:/GPT/ OR (intitle:/\bOpen[ ]?AI\b/i) OR (#/\bOpen[ ]?AI\b/i) OR intitle:/\bAnthropic\b/i OR intitle:/\bAltman\b/i OR (intitle:/\bChat[ ]?bots?\b/i) OR (intitle:/\btraining data(sets?)\b/i) OR intitle:/\bMidjourney\b/i OR intitle:/\bA[.]?I[.]?\b/i OR #/\bA[.]?I[.]?\b/i OR #/\bM[.]?L[.]?\b/i OR (#"Machine Learning") OR intitle:/\bM[.]?L[.]?\b/i OR intitle:/vibe\s?cod(e|ing)/i OR #/vibe\s?cod(e|ing)/i OR "intilte:/Deep\s?[Ss]eek/" OR intitle:/\bM[.]?C[.]?P[.]?\b/ OR #/\bM[.]?C[.]?P[.]?\b/ OR /\bagentic\b/i
```

EDIT: autocorrect hyphenation

**Reply by mtalexan** [reply]
To clarify some of these:
- Art is things like art shows from local news sources, or the NYT, or similar.
- Death is for announcements of people dying, and the follow up "in memory" articles
- Financial is for filtering sources like TechMeme that insist on giving the blow by blow of funding rounds and quarterly financial statements
- InReview is for the articles everyone puts out at the end of every year and start of the next rehashing what happened in the last year or decade or whatever.
- MeetsVisits is for the news announcements about a government official visiting another country or person or whatever
- History comes up in local news sources and NPR a lot, but is self-explanatory from the filter
- Timeline used to be more complicated, but is for those pieces that rehash what's been happening in a major disaster, war, etc once the new information slows down and they run out of things to write about
- AI is what you think. It's one that I constantly have to add to since I try to add the names of new popular models to it when they start slipping thru. 

I also have a slew of filters specific to individual sources. I'll often have multiple focused feeds from the same source so I created "Common*" filters for them and reference it on each feed so I can update it in just one place.

**Reply by mtalexan** [reply]
Worth noting too, these can also be used to assign labels instead/as well if you wanted. The AI one for example I filter out on some feeds, but I also have a global Label assignment using it so the stuff from feeds I do want to hear about AI from gets properly labeled.

**Comment by zapovit** [comment]
*filter for sports*
```
inurl:/\b(?<!tran)sport(s)?\b/
```
Will match any url containing sport or sports but NOT transport(s) (i.e. trucking, drayage, etc.) `\b` will match -sport-. You could also use this as `intitle:` or free text search, as `\b` also captures " sport " but probably want to make it case insensitive for those two.

**Reply by Alkarex** [reply]
Or maybe just `inurl:/\bsports?\b/i` ?

**Reply by zapovit** [reply]
Oops. Yeah, guess the negative lookahead and word boundaries don't make much sense together. Good observation.

**Comment by zapovit** [comment]
I made a pretty good user query to capture podcasts which Bloomberg and others seem to include in many opinion/commentary feeds:
```
(inurl:/\/audio\// OR inurl:/\/podcast(s)?\// OR intitle:/podcast\:/i) (pubdate:P3M)
```
I include the pubdate for the purpose of resyndicating the feed but you may wish to remove it if you use user queries to "mark as read"...
